airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -206,7 +206,7 @@ def _group_streams(
             # so we need to treat them as synchronous
             if (
                 isinstance(declarative_stream, DeclarativeStream)
-                and name_to_stream_mapping[declarative_stream.name].get("retriever")["type"]
+                and name_to_stream_mapping[declarative_stream.name]["retriever"]["type"]
```

Comment:
Note: Previous code was unsafe because if "retriever" is not present during "get()", then "None" would not be subscriptable. This way, return type is "Any" which doesn't upset the typer - but of course the safer way is to check types before we infer that they are present - or just noqa it with the assumption we are making on the structure.